### PR TITLE
Implement data collection and analytics

### PR DESCRIPTION
**Pull request by katlyn**

### CODE REVIEW

**Review by cyyynthia — APPROVED** [review]

**Comment by aetheryx** [comment]
hmm

File: src/Powercord/apis/analytics.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    // The analytics API does not unload.
+    this.warn('Analytics cannot be unloaded.');
+  }
+
+  _sendAnalyticsData (data) {
+    this.log(`Starting analytics submissions for ${data.type}`);
+    this.endpoints.forEach((endpoint, i) => {
+      post(endpoint).send(data)
+        .then(() => this.log(`Submitted analytics data for ${i + 1}/${this.endpoints.length} endpoints`))
+        .catch(() => this.warn(`Unable to submit analytics data to ${endpoint}! Attempting to block analytics collection is in direct violation of our Terms of Service and Privacy Policy and my result in termination of support.`));
```
Comment (Nevvulo) [inline src/Powercord/apis/analytics.js:54]: Typo - "and my result"

**Review by Nevvulo — CHANGES_REQUESTED** [review]

File: src/Powercord/apis/analytics.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  apiWillUnload () {
+    // The analytics API does not unload.
+    this.warn('Analytics cannot be unloaded.');
+  }
+
+  _sendAnalyticsData (data) {
+    this.log(`Starting analytics submissions for ${data.type}`);
+    this.endpoints.forEach((endpoint, i) => {
```
Comment (MrJacz) [inline src/Powercord/apis/analytics.js]: Can I suggest using .map and Promise.all

Comment (aetheryx) [reply src/Powercord/apis/analytics.js]: why would you use map and promise.all for something that doesn't need to be awaited

**Comment by ghostlydilemma** [comment]
Can we finally merge this i mean common i have been waiting for like half a year now

**Pull request closed without merging**